File: examples/typescript/react/notes-showcase/src/models/seed.ts
/**
 * Deterministic cross-language seed data.
 *
 * Mirrors the C# `SeedData` / Python `build_seed` content (same notebook ids,
 * note ids, and starred flags) so cross-language parity audits compare
 * identically.
 */
import type { NotebookModel } from "./notebookModel.js";
import type { NoteModel } from "./noteModel.js";

const NOW_MS = Date.UTC(2026, 4, 29, 12, 0, 0); // 2026-05-29T12:00:00Z

const HOUR_MS = 60 * 60 * 1000;
const DAY_MS = 24 * HOUR_MS;

function iso(ms: number): string {
  return new Date(ms).toISOString();
}

/**
 * Returns the canonical seed: 5 notebooks (1 nested), 12 notes, 3 starred.
 *
 * Returned arrays are frozen so callers cannot mutate the seed source.
 */
export function buildSeed(): {
  notebooks: readonly NotebookModel[];
  notes: readonly NoteModel[];
} {
  const notebooks: NotebookModel[] = [
    { id: "nb-work", name: "Work", parentId: null },
    { id: "nb-specs", name: "Specs", parentId: "nb-work" },
    { id: "nb-reviews", name: "Reviews", parentId: null },
    { id: "nb-personal", name: "Personal", parentId: null },
    { id: "nb-archive", name: "Archive", parentId: null },
  ];

  const notes: NoteModel[] = [];
  let idx = 0;

  function add(
    notebookId: string,
    title: string,
    opts: { starred?: boolean; tags?: readonly string[] } = {},
  ): void {
    idx += 1;
    notes.push({
      id: `note-${String(idx).padStart(2, "0")}`,
      notebookId,
      title,
      tags: Object.freeze([...(opts.tags ?? [])]),
      body: `(seed body for ${title})`,
      starred: opts.starred ?? false,
      createdAt: iso(NOW_MS - idx * DAY_MS),
      updatedAt: iso(NOW_MS - idx * HOUR_MS),
    });
  }

  add("nb-reviews", "Q1 design review");
  add("nb-reviews", "Auth migration plan", { starred: true, tags: ["security", "q2"] });
  add("nb-reviews", "Vendor shortlist");
  add("nb-reviews", "Onboarding draft");
  add("nb-reviews", "Privacy review notes");
  add("nb-reviews", "Disaster recovery plan");
  add("nb-reviews", "Cross-team review log", { starred: true });
  add("nb-work", "Standup notes");
  add("nb-work", "Roadmap snapshot");
  add("nb-specs", "MVx capability brief", { starred: true });
  add("nb-personal", "Reading list");
  add("nb-personal", "Travel ideas");

  return {
    notebooks: Object.freeze(notebooks),
    notes: Object.freeze(notes),
  };
}

File: examples/python/textual/notes_showcase/src/notes_showcase/models/seed.py
"""Deterministic cross-language seed data.

Mirrors the C# ``SeedData`` content (same notebook ids, note ids, and starred
flags) so cross-language audits compare identically. Data-only — excluded from
the coverage threshold via pytest configuration.
"""

from __future__ import annotations

from datetime import datetime, timedelta, timezone

from notes_showcase.models.note_model import NoteModel
from notes_showcase.models.notebook_model import NotebookModel


def build_seed() -> tuple[list[NotebookModel], list[NoteModel]]:
    """Return the canonical seed: 5 notebooks (1 nested), 12 notes, 3 starred."""
    # Deterministic "now" so cross-language hash comparisons are stable.
    now = datetime(2026, 5, 29, 12, 0, 0, tzinfo=timezone.utc)

    notebooks: list[NotebookModel] = [
        NotebookModel(id="nb-work", name="Work", parent_id=None),
        NotebookModel(id="nb-specs", name="Specs", parent_id="nb-work"),
        NotebookModel(id="nb-reviews", name="Reviews", parent_id=None),
        NotebookModel(id="nb-personal", name="Personal", parent_id=None),
        NotebookModel(id="nb-archive", name="Archive", parent_id=None),
    ]

    notes: list[NoteModel] = []
    idx = 0

    def add(
        notebook_id: str,
        title: str,
        *,
        starred: bool = False,
        tags: tuple[str, ...] = (),
    ) -> None:
        nonlocal idx
        idx += 1
        notes.append(
            NoteModel(
                id=f"note-{idx:02d}",
                notebook_id=notebook_id,
                title=title,
                tags=tags,
                body=f"(seed body for {title})",
                starred=starred,
                created_at=now - timedelta(days=idx),
                updated_at=now - timedelta(hours=idx),
            )
        )

    add("nb-reviews", "Q1 design review")
    add("nb-reviews", "Auth migration plan", starred=True, tags=("security", "q2"))
    add("nb-reviews", "Vendor shortlist")
    add("nb-reviews", "Onboarding draft")
    add("nb-reviews", "Privacy review notes")
    add("nb-reviews", "Disaster recovery plan")
    add("nb-reviews", "Cross-team review log", starred=True)
    add("nb-work", "Standup notes")
    add("nb-work", "Roadmap snapshot")
    add("nb-specs", "MVx capability brief", starred=True)
    add("nb-personal", "Reading list")
    add("nb-personal", "Travel ideas")

    return notebooks, notes
